### PR TITLE
[ADD] solution hw-7.1

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/core/basesyntax/Pair.java [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        Pair<T, V> pair = (Pair<T, V>) object;
+        return ((val1 == pair.val1) || (val1 != null && val1.equals(pair.val1)))
+                && (val2 == pair.val2) || (val2 != null && val2.equals(pair.val2));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Pair.java]: review your parenthesis please

File: src/main/java/core/basesyntax/Pair.java [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+    private V val2;
 
+    private Pair(T val1, V val2) {
+        this.val1 = val1;
+        this.val2 = val2;
+    }
+
+    public T getFirst() {
+        return val1;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Pair.java]:
```suggestion
    public T getFirst() {
        return first;
    }
```
what do you think?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Pair.java [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        Pair<T, V> pair = (Pair<T, V>) object;
+        return ((first == pair.first) || first != null && first.equals(pair.first)
+                && (second == pair.second) || (second != null && second.equals(pair.second)));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Pair.java]: at the moment this part `first != null && first.equals(pair.first) && (second == pair.second)` is evaluated together

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Pair.java (hunk 3 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Pair.java]: done

File: src/main/java/core/basesyntax/Pair.java (hunk 2 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Pair.java]: agree, fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**